app/models/host.rb

```diff
@@ -1413,6 +1351,69 @@ def scan_from_queue(taskid = nil)
     _log.info("Scanning #{log_target}...Complete - Timings: #{t.inspect}")
   end
 
+  def scan_via_ssh(task)
+    if hostname.blank?
+      _log.warn("No hostname defined for #{log_target}")
+      task.update_status("Finished", "Warn", "Scanning incomplete due to missing hostname")  if task
+      return
+    end
+
+    update_ssh_auth_status! if respond_to?(:update_ssh_auth_status!)
+
+    if missing_credentials?
```

Comment:
openstack infra also overrides this method

Reply:
I am comfortable with overrides

Reply:
Yeah I'm just trying to figure out how much of this is actually just OpenStack specific, only other provider which this would apply to is Ovirt

Reply:
Is all of this prelude just verifying that the SSH auth is valid?  If so, and we do https://github.com/ManageIQ/manageiq/pull/20802/files#r521617144, then this all feels greatly simplified by pushing it down into verifying an SSH key pair.